test/e2e/thanos_querier_test.go

```diff
@@ -91,3 +92,137 @@ func TestThanosQueryCanQueryWatchdogAlert(t *testing.T) {
+		},
+		{
+			name:      fmt.Sprintf("assert %s cluster role exists", clusterRoleName),
+			assertion: f.AssertClusterRoleExists(clusterRoleName),
+		},
+		{
+			name:      fmt.Sprintf("thanos querier API is accessible by cluster role %s", clusterRoleName),
+			assertion: testClusterRoleAccessToThanosQuerier,
+		},
+		{
+			name:      fmt.Sprintf("thanos querier API is accessible by role %s", roleName),
+			assertion: testRoleAccessToThanosQuerier,
+		},
+		{
+			name:      "thanos querier API is inaccessible without either role or cluster role",
+			assertion: testBlockedAccessToThanosQuerier,
+		},
+	} {
+		t.Run(tc.name, tc.assertion)
+	}
+}
+
+func testClusterRoleAccessToThanosQuerier(t *testing.T) {
+	const clusterRoleName = "cluster-monitoring-view"
+	const saName = "e2e-thanos-querier"
+
+	cleanUpFuncs := []framework.CleanUpFunc{}
+	cf, err := f.CreateServiceAccount(saName, f.Ns)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cleanUpFuncs = append(cleanUpFuncs, cf)
+
+	cf, err = f.CreateClusterRoleBinding(f.Ns, saName, clusterRoleName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cleanUpFuncs = append(cleanUpFuncs, cf)
+
+	var lastError error
+	err = wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 2*time.Minute, true, func(context.Context) (bool, error) {
+		_, err = f.ThanosQuerierClient.PrometheusTargets()
+		if err != nil {
+			lastError = err
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatal(lastError)
+	}
+
+	for _, cf := range cleanUpFuncs {
+		err := cf()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+}
+
+func testRoleAccessToThanosQuerier(t *testing.T) {
+	const roleName = "cluster-monitoring-metrics-api"
+	const saName = "e2e-thanos-querier"
+	cleanUpFuncs := []framework.CleanUpFunc{}
+
+	cf, err := f.CreateServiceAccount(saName, f.Ns)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cleanUpFuncs = append(cleanUpFuncs, cf)
+
+	cf, err = f.CreateRoleBindingFromRole(f.Ns, saName, roleName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cleanUpFuncs = append(cleanUpFuncs, cf)
+
+	var lastError error
+	err = wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 2*time.Minute, true, func(context.Context) (bool, error) {
+		_, err = f.ThanosQuerierClient.PrometheusTargets()
+		if err != nil {
+			lastError = err
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatal(lastError)
+	}
+
+	for _, cf := range cleanUpFuncs {
+		err := cf()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+}
+
+func testBlockedAccessToThanosQuerier(t *testing.T) {
+	const saName = "e2e-thanos-querier"
+	cleanUpFuncs := []framework.CleanUpFunc{}
+
+	cf, err := f.CreateServiceAccount(saName, f.Ns)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cleanUpFuncs = append(cleanUpFuncs, cf)
+
+	err = wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 2*time.Minute, true, func(context.Context) (bool, error) {
+		_, err = f.ThanosQuerierClient.PrometheusTargets()
```

Comment:
I'd use f.ThanosQuerierClient..PrometheusQueryWithStatus() and assert that the expected HTTP code is returned (403?).